mentors/Makefile

```diff
@@ -0,0 +1,64 @@
+DOCKER_IMAGE?=mentor-processing-pipeline
+DOCKER_CONTAINER=mentor-processing-pipeline
+MENTOR?=julianne
+PIPELINE_PATH=src/data_pipeline/pipeline_entrypoint.py
+SESSION?=1
+
+# Removes all mentor files from the local file system
+.PHONY clean:
+clean:
+	rm -rf recordings/*
+
+# Builds the data processing pipeline dockerfile
+docker-build:
+	docker build -t $(DOCKER_IMAGE) .
+
+# Runs a shell inside the data processing pipeline dockerfile
+docker-run:
+	docker run \
+			-it \
+			--rm \
+			--name $(DOCKER_CONTAINER) \
+			-v $(shell pwd)/recordings:/app/recordings \
+			$(DOCKER_IMAGE) \
+			/bin/bash
+
+# Fully pre-processes mentor data inside docker container
+# Generates audiochunks and transcripts for all sessions
+docker-build-mentor:
+	docker run \
+			-it \
+			--rm \
+			--name $(DOCKER_CONTAINER) \
+			-v $(shell pwd)/recordings:/app/recordings \
+			--entrypoint python3 $(DOCKER_IMAGE) $(PIPELINE_PATH) -m $(MENTOR) -t
+
+# Partially pre-processes mentor data inside docker container
+# Generates only audiochunks for all sessions
+docker-build-audiochunks:
+	docker run \
+			-it \
+			--rm \
+			--name $(DOCKER_CONTAINER) \
+			-v $(shell pwd)/recordings:/app/recordings \
+			--entrypoint python3 $(DOCKER_IMAGE) $(PIPELINE_PATH) -m $(MENTOR)
+
+# Partially pre-processes mentor data inside docker container
+# Generates audiochunks and transcripts for one session
+docker-build-session:
```

Comment:
for all the docker operations, since they're the main event, maybe we can simplify the names and drop the 'docker-` prefix?

Also, why wouldn't `src/data_pipeline/pipeline_entrypoint.py` be the default entry point? and then each distinct operation can just pass a command arg?

Reply:
I can do that, but I thought that there may be cases where someone wants to manually step into the container (whoever has to maintain these processes). In `docker-run`, I just set the entry point to `/bin/bash`. 

Also we can't pass an ARG to ENTRYPOINT. We could use an ARG to set an ENV and pass that to the ENTRYPOINT though. What do you think?

Reply:
yeah, the case of overriding the entrypoint to `/bin/bash` still works for `shell`, so that should be like this in the makefile:

```make
shell:
	docker run \
			-it \
			--rm \
			--name $(DOCKER_CONTAINER) \
                         --entrypoint /bin/bash \
			-v $(shell pwd)/recordings:/app/recordings \
			$(DOCKER_IMAGE) \
			/bin/bash
```

For everthing else (in docker terms), `python src/data_pipeline/pipeline_entrypoint.py` is the `ENTRYPOINT` and whatever you pass to it is the `CMD`

So the dockerfile gets

```Dockerfile
ENTRYPOINT ["python", "src/data_pipeline/pipeline_entrypoint.py"]
```

...and an example make rule would be like

```make
build-session:
	docker run \
			-it \
			--rm \
			--name $(DOCKER_CONTAINER) \
			-v $(shell pwd)/recordings:/app/recordings \
			$(DOCKER_IMAGE) "-m $(MENTOR) -s $(SESSION) -t"
```